File: backend/internal/models/dto/statistics.go
package dto

import "time"

type ShortNovelaInfo struct {
	NovelaID    int			`json:"novelaId"`
	Title       string		`json:"title"`
	Description string		`json:"description"`
	PosterURL   string		`json:"posterUrl"`
	ReleaseDate time.Time	`json:"releaseDate"`
}

type TotatStatistics struct {
	Novela ShortNovelaInfo `json:"novela"`

	TotalBookmarkCount int     `json:"bookmarkCount"`
	TotalReadCount     int     `json:"readCount"`
	TotalRatingCount   int     `json:"ratingCount"`
	TotalCommentCount  int     `json:"commentCount"`
	TotalVolumeCount   int     `json:"volumeCount"`
	TotalChapterCount  int     `json:"chapterCount"`
	Rating             float64 `json:"rating"`
}

type TotalStatisticsResponse struct {
	Data  *[]TotatStatistics `json:"data"`
	Total int                `json:"total"`
}

type GeneralStatistics struct {
	BookmarkCount int `json:"bookmarkCount"`
	ReadCount     int `json:"readCount"`
	RatingCount   int `json:"ratingCount"`
	CommentCount  int `json:"commentCount"`
	VolumeCount   int `json:"volumeCount"`
	ChapterCount  int `json:"chapterCount"`
}

type ShortNovelaMonthly struct {
	NovelaID     int    `json:"novelaId"`
	Title        string `json:"title"`
	PosterURL    string `json:"posterUrl"`
}

type NovelaMonthlySeries struct {
	Novela       ShortNovelaMonthly `json:"novela"`
	MonthlyReads []int              `json:"monthlyReads"`
}

File: frontend/app/types/backend/statistics.ts
export interface NovelaShort {
  novelaId: number;
  title: string;
  description: string;
  posterUrl: string;
  releaseDate: Date;
}

export interface TotalNovelaStatistics {
  novela: NovelaShort;
  bookmarkCount: number;
  readCount: number;
  ratingCount: number;
  commentCount: number;
  volumeCount: number;
  chapterCount: number;
  rating: number;
}

export interface TotalNovelaStatisticsResponse {
  data: TotalNovelaStatistics[];
  total: number;
}

export interface GeneralStatistics {
  bookmarkCount: number;
  readCount: number;
  ratingCount: number;
  commentCount: number;
  volumeCount: number;
  chapterCount: number;
}

export interface MonthlyStatisticsItem {
  month: number;
  year: number;
  bookmarkCount: number;
  readCount: number;
  commentCount: number;
  ratingCount: number;
}

interface ShortNovelaMonthly {
  novelaId: number;
  title: string;
  posterUrl: string;
}

export interface NovelaMonthlySeries {
  novela: ShortNovelaMonthly;
  monthlyReads: number[];
}
